File: frontend/src/app/types/pagamentos.ts
export type LigaCobrancaMesResponse = {
  id: number;
  liga_id: number;
  mes: number;
  ativo: boolean;
  updated_at: string;
};

export type LigaPagamentoResponse = {
  id: number;
  liga_id: number;
  usuario_id: number;
  mes: number;
  pago: boolean;
  updated_at: string;
  updated_by?: number | null;
};

export type LigaPagamentoUpsert = {
  mes: number;
  pago: boolean;
};

File: backend/app/schemas/pagamentos.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class LigaPagamentoResponse(BaseModel):
    id: int
    liga_id: int
    usuario_id: int
    mes: int = Field(..., ge=1, le=12)
    pago: bool
    updated_at: datetime
    updated_by: Optional[int] = None

    class Config:
        from_attributes = True


class LigaPagamentoUpsert(BaseModel):
    mes: int = Field(..., ge=1, le=12)
    pago: bool
